### PR TITLE
Add search to game list

### LINKED ISSUE #750: Games/List - Add Search
particularly for systems like NES: https://tasvideos.org/Games/List?SystemCode=NES

We need a way to search game names here.  Finding something in the middle letters of the alphabet is a drag

### PR DESCRIPTION
**Pull request by Masterjun3**
Resolves #750.

Not really happy with this. Technically I don't need to create an own "SearchTerms" variable in /Games/List.cshtml.cs, the query automatically takes it to the "Search" and into the "(Search.)SearchTerms". But now the URL will say "Search.SearchTerms" which breaks things, so it's possible to overwrite the "name" attribute inside the .cshtml. However, this causes the validation to not work, because it relies on the attribute.

In short we either have no automatic validation, or an extra variable in the root like this.

### CODE REVIEW

File: TASVideos/Pages/Games/List.cshtml.cs [hunk 1]
```diff
@@ -34,7 +41,12 @@ public class ListModel : BasePageModel
 
 		public async Task OnGet()
 		{
-			Games = await GetPageOfGames(Search);
+			if (ModelState.IsValid)
```
Comment (adelikat) [inline TASVideos/Pages/Games/List.cshtml.cs:44]: Why are we checking the modelstate on the get method?

Comment (Masterjun3) [reply TASVideos/Pages/Games/List.cshtml.cs:44]: The ModelState is invalid when the search terms of the url query aren't satisfying the conditions (which are: string must be empty or minimum of 5 characters)

**Review by adelikat — APPROVED** [review]

File: TASVideos/Pages/Games/List.cshtml.cs (hunk 1 above)
Comment (adelikat) [reply TASVideos/Pages/Games/List.cshtml.cs:44]: That makes sense, just an unusual pattern, but I think this is the simplest way to handle it
